src/formElementPlugins/SummaryViewWrapper.tsx

```diff
@@ -1,12 +1,58 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { ErrorBoundary, pluginProvider } from '.'
+import { Grid, Icon } from 'semantic-ui-react'
+import { SummaryViewWrapperProps, PluginComponents, ValidationState } from './types'
 import { Form, Header } from 'semantic-ui-react'
-import { SummaryViewWrapperProps, PluginComponents } from './types'
 import { TemplateElementCategory } from '../utils/generated/graphql'
+import { defaultValidate } from './defaultValidate'
+import { EvaluatorParameters } from '../utils/types'
+import { IQueryNode } from '@openmsupply/expression-evaluator/lib/types'
 
 const SummaryViewWrapper: React.FC<SummaryViewWrapperProps> = (props) => {
-  const { element, response } = props
+  const { element, response, allResponses, isStrictValidation } = props
   const { parameters, category, code, pluginCode, isEditable, isRequired, isVisible } = element
+  const { validation: validationExpression, validationMessage } = parameters
+  const [validationState, setValidationState] = useState<ValidationState>({} as ValidationState)
+  const responses = { thisResponse: response?.text, ...allResponses }
+  const [pluginMethods, setPluginMethods] = useState({
+    validate: (
+      validationExpress: IQueryNode,
+      validationMessage: string,
+      evaluatorParameters: EvaluatorParameters
+    ): any =>
+      defaultValidate(validationExpression, validationMessage, {
+        objects: [responses],
+        APIfetch: fetch,
+      }),
+  })
+
+  useEffect(() => {
+    if (!pluginCode) return
+    // TODO use plugin-specific validation method if defined
+    setPluginMethods({
+      validate: defaultValidate,
+    })
+  }, [])
+
+  useEffect(() => {
+    // Re-validate on load
+    if (!isStrictValidation && (!validationExpression || response?.text === undefined)) {
+      setValidationState({ isValid: true } as ValidationState)
+      return
+    }
+
+    if (isStrictValidation && isRequired && response?.text === undefined) {
+      console.log('Going to invalid:', code)
+      setValidationState({ isValid: false, validationMessage: 'Field cannot be blank' })
+      return
+    }
+
+    pluginMethods
+      .validate(validationExpression, validationMessage, { objects: [responses], APIfetch: fetch })
+      .then((validationResult: ValidationState) => {
+        setValidationState(validationResult)
+      })
+  }, [isStrictValidation])
```

Comment:
That's cool. So this is doing a `strict` validation without sending the data to the server. Which I think would prevent the application to show all problems when trying to go back to the application page. Is that correct?